airsonic-main/src/main/java/org/airsonic/player/service/AudioScrobblerService.java

```diff
@@ -18,309 +18,60 @@
-        params.put("s", sessionId);
-        params.put("a", registrationData.artist);
-        params.put("t", registrationData.title);
-        params.put("b", registrationData.album);
-        params.put("l", String.valueOf(registrationData.duration));
-        params.put("n", "");
-        params.put("m", "");
-        return executePostRequest(url, params);
-    }
-
-    private String calculateAuthenticationToken(String password, long timestamp) {
-        return DigestUtils.md5Hex(DigestUtils.md5Hex(password) + timestamp);
-    }
-
-    private String[] executeGetRequest(URI url) throws IOException {
-        HttpGet method = new HttpGet(url);
-        method.setConfig(requestConfig);
-        return executeRequest(method);
-    }
-
-    private String[] executePostRequest(String url, Map<String, String> parameters) throws IOException {
-        List<NameValuePair> params = new ArrayList<NameValuePair>();
-        for (Map.Entry<String, String> entry : parameters.entrySet()) {
-            params.add(new BasicNameValuePair(entry.getKey(), entry.getValue()));
-        }
-
-        HttpPost request = new HttpPost(url);
-        request.setEntity(new UrlEncodedFormEntity(params, StringUtil.ENCODING_UTF8));
-        request.setConfig(requestConfig);
-        return executeRequest(request);
-    }
-
-    private String[] executeRequest(HttpUriRequest request) throws IOException {
-
-        try (CloseableHttpClient client = HttpClients.createDefault()) {
-            ResponseHandler<String> responseHandler = new BasicResponseHandler();
-            String response = client.execute(request, responseHandler);
-            return response.split("\\n");
-
-        }
-    }
-
-    public void setSettingsService(SettingsService settingsService) {
-        this.settingsService = settingsService;
-    }
-
-    private class RegistrationThread extends Thread {
-        private RegistrationThread() {
-            super("AudioScrobbler Registration");
-        }
-
-        @Override
-        public void run() {
-            while (true) {
-                RegistrationData registrationData = null;
-                try {
-                    registrationData = queue.take();
-                    scrobble(registrationData);
-                } catch (IOException x) {
-                    handleNetworkError(registrationData, x);
-                } catch (Exception x) {
-                    LOG.warn("Error in Last.fm registration.", x);
-                }
+        UserSettings userSettings = settingsService.getUserSettings(username);
+        if (userSettings.isLastFmEnabled() && userSettings.getLastFmUsername() != null && userSettings.getLastFmPassword() != null) {
+            if (lastFMScrobbler == null) {
+                lastFMScrobbler = new LastFMScrobbler();
             }
+            lastFMScrobbler.register(mediaFile, userSettings.getLastFmUsername(), userSettings.getLastFmPassword(), submission, time);
```

Comment:
It would be better to pass uername and password through the constructor.

Reply:
Is this service construct one for every users? I was guessing that this service object shared between all users, so I passed this data through register function.

Reply:
Ah, I see, if you see it as a long-running service that's shared for everyone it makes sense.

What do you think of turning it into a real Spring service then? IMO this makes it more clear that it's intended this way and avoids the `if (lastFMScrobbler == null)` dance a few lines above.